rotkehlchen/chain/evm/decoding/balancer/decoder.py

```diff
@@ -0,0 +1,220 @@
+from abc import ABC
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Final, Literal
+
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.balancer.balancer_cache import query_balancer_data
+from rotkehlchen.chain.evm.decoding.balancer.constants import CPT_BALANCER_V1, CPT_BALANCER_V2
+from rotkehlchen.chain.evm.decoding.interfaces import (
+    DecoderInterface,
+    ReloadablePoolsAndGaugesDecoderMixin,
+)
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import CacheType, ChainID, ChecksumEvmAddress, EvmTransaction
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+GAUGE_DEPOSIT_TOPIC: Final = b'\xe1\xff\xfc\xc4\x92=\x04\xb5Y\xf4\xd2\x9a\x8b\xfcl\xda\x04\xeb[\r<F\x07Q\xc2@,\\\\\xc9\x10\x9c'  # noqa: E501
+GAUGE_WITHDRAW_TOPIC: Final = b'\x88N\xda\xd9\xceo\xa2D\r\x8aT\xcc\x124\x90\xeb\x96\xd2v\x84y\xd4\x9f\xf9\xc76a%\xa9BCd'  # noqa: E501
+
+
+class BalancerCommonDecoder(DecoderInterface, ReloadablePoolsAndGaugesDecoderMixin, ABC):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            pool_cache_type: CacheType,
+            read_fn: Callable[[ChainID], tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]],
+            counterparty: Literal['balancer-v1', 'balancer-v2'],
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        ReloadablePoolsAndGaugesDecoderMixin.__init__(
+            self,
+            evm_inquirer=evm_inquirer,
+            cache_type_to_check_for_freshness=pool_cache_type,
+            query_data_method=query_balancer_data,
+            read_data_from_cache_method=read_fn,
+            chain_id=evm_inquirer.chain_id,
+        )
```

Comment:
is this the only new thing? `_decode_gauge_events` and `_check_deposits_withdrawals` do they have changes?

Reply:
the only new thing is `_decode_gauge_events()`